### DISCUSSION TITLE
Update on timeline or roadmap

### DISCUSSION THREAD

**Discussion by mrob1600** [dotnet/yarp#651] [General] [answered]
Just wondering if you have an update on a rough estimate or timeline for when we could expect 1.0. I thought I saw mentioned somewhere that it could be not too long after the net5.0 release, but I probably misread that. Just asking for rough planning purposes on our side.

**Comment by Tratcher** [comment]
We couldn't release any earlier than .NET 5.0.

We're working with a few early adopters to get them running and deployed. Only when that's successful will we plan to declare a 1.0 version. It will likely be a few more months.

Have you been using the previews? Is there anything you'd need before 1.0?

**Reply by mrob1600** [reply]
Thank you for the update. 

Yes, we have been running this since around preview 3 for an app that is under development - a collection of net5.0 services and an angular client. As of preview 8 it seems the features we actually need are there. As more features are added we would be able to take advantage of those but don't need them at this time. 

Also, enough functionality is there as for us to begin replacing some of the endpoints from our F5 BigIP devices with YARP (once it is production ready of course) so that we can eventually retire those devices, as they are overkill for our use cases (can live without layer 4 and payload manipulation), complex to manage and support, and we are tired of writing and maintaining iRules using Tcl, etc.

**Reply by dominikjeske** [reply]
Do you have some ideas on when 1.0 will be shipped. We can test on preview but we cannot go to production with those bits so some timeline would be great. It is important to know is 1.0 will arrive in Q2, Q3 or Q4?

**Reply by Tratcher** [reply]
When do you need it?

**Reply by dominikjeske** [reply]
I think that Q2 for dev/testing and Q3 for production. Can we count on that?

**Reply by Tratcher** [reply]
We already have some early adopters deployed to production and are mainly waiting on their sign-off before we ship. Q3 should not be a problem.

**Reply by arunravir** [reply]
Can we count on LTS version for beginning of Q3 ?

**Reply by Tratcher** [reply]
@arunravir I can't be more specific about a release date at this time.

Note we haven't outlined the LTS policy yet, the conversation so far is only about the 1.0 release date.

**Comment by karelz** [comment, marked as answer]
Roadmap has been published: https://github.com/microsoft/reverse-proxy/blob/main/docs/roadmap.md
